sdk/storage/azure-storage-blob/tests/perfstress_tests/_test_base.py

```diff
@@ -32,8 +33,28 @@ def __init__(self, arguments):
         # self._client_kwargs['api_version'] = '2019-02-02'  # Used only for comparison with T1 legacy tests
 
         if not _ServiceTest.service_client or self.args.no_client_share:
-            _ServiceTest.service_client = SyncBlobServiceClient.from_connection_string(conn_str=connection_string, **self._client_kwargs)
-            _ServiceTest.async_service_client = AsyncBlobServiceClient.from_connection_string(conn_str=connection_string, **self._client_kwargs)
+            if self.args.use_entra_id:
+                tenant_id = self.get_from_env("AZURE-STORAGE-BLOB_TENANT_ID")
```

Comment:
Let's also add a bit in the perfstress_tests/README.md file in the `Setup for test resources` section about what env vars need to be set when using Entra ID authentication (i.e. tenant_id, client_id, client_secret, account_name env vars).

Reply:
Out of curiosity, how does the pipeline handle these new variables? What AAD app is it using or does it make one each time?

Reply:
It uses the default ones that are deployed once per run (which are deployed by default for all pipelines with <SERVICE>_TENANT_ID/CLIENT_ID/CLIENT_SECRET). https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3669358&view=logs&j=42ded549-05ee-5cdb-7ba7-7a948a0cc056&t=1a18f1cb-d016-5e13-bec4-a5dc2ac35679&l=74

For the account name, it's the one that's already included in the perf-resources.bicep, so we didn't need to add anything new there.